Tutorial/heat_transfer/io_adios2.F90

```diff
@@ -0,0 +1,114 @@
+!  ADIOS is freely available under the terms of the BSD license described
+!  in the COPYING file in the top level directory of this source distribution.
+!
+!  Copyright (c) 2008 - 2017.  UT-BATTELLE, LLC. All rights reserved.
+!  ADIOS2 I/O for the heat_transfer example
+!
+! (c) Oak Ridge National Laboratory, 2018
+! Author: William F Godoy
+!
+module heat_io
+
+    implicit none
+    ! adios handlers
+    integer*8 :: adios, io, bp_writer, var_gndx, var_gndy, var_T, var_dT
+
+contains
+
+subroutine io_init()
+    use heat_vars
+    use adios2
+    implicit none
+
+    call adios2_init (adios, app_comm, adios2_debug_mode_on, ierr)
+    call adios2_declare_io (io, adios, 'heat', ierr )
+end subroutine io_init
+
+subroutine io_finalize()
+    use heat_vars
+    use adios2
+    implicit none
+
+    call adios2_close(bp_writer, ierr)
+    call adios2_finalize (adios, ierr)
+end subroutine io_finalize
+
+subroutine io_write(tstep,curr)
+    use heat_vars
+    use adios2
+    implicit none
+    include 'mpif.h'
+    integer, intent(in) :: tstep
+    integer, intent(in) :: curr
+
+    integer*8 :: adios_handle, adios_totalsize
+    integer :: adios2_err
+    character (len=200) :: filename
+    character(2) :: mode = "w"
+    ! variables for definition
+    integer*8 :: varid
+    ! character(len=100) :: gdims, ldims, offs
+    ! in this case shape = gdims, start = offs, count = ldims
+    integer*8, dimension(2) :: shape_dims, start_dims, count_dims
+
+    write(filename,'(a,".bp")') trim(outputfile)
+    if (rank==0.and.tstep==1) then
+        print '("Writing: "," filename ",14x,"size(GB)",4x,"io_time(sec)",6x,"GB/s")'
+    endif
+
+    ! if (tstep > 1) mode = "a"
+
+    ! Define variables at the first time
+    if (tstep.eq.1) then
+
+        call adios2_open (bp_writer, io, filename, adios2_mode_write, adios2_err)
```

Comment:
You are not passing the communicator to `adios2_open`. I see that `adios2_open` is an interface to two subroutines ` adios2_open_old_comm` and `adios2_open_new_comm`. I assume that if you dont pass the communicator, it uses the one that was used in `adios2_init` ?

Reply:
@kshitij-v-mehta that's correct, indeed. http://adios2-adaptable-io-system-version-2.readthedocs.io/en/latest/api/api.html#opening-an-engine